hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -1321,7 +1321,18 @@ public Option<HoodieFileFormat> getPartitionMetafileFormat() {
   }
 
   public Map<String, String> getTableMergeProperties() {
-    return ConfigUtils.extractWithPrefix(this.props, RECORD_MERGE_PROPERTY_PREFIX);
+    Map<String, String> configs = ConfigUtils.extractWithPrefix(this.props, RECORD_MERGE_PROPERTY_PREFIX);
+    if (getTableVersion().lesserThan(HoodieTableVersion.NINE)) {
```

Comment:
The write path uses the `ConfigUtils.getMergeProps` instead, while this is just for read path, should we just all invoke `ConfigUtils.getMergeProps` instead.